### DISCUSSION TITLE
vim's `make`, `cnext`, `cprev` (or Emacs' `compile`, `next-error`, `previous-error`)

### DISCUSSION THREAD

**Discussion by leungbk** [helix-editor/helix#4290] [Ideas] [answered]
I cannot find any analogues of these commands in the command list for helix. At least in Emacs (I am not experienced with Vim), upon a compilation error, you can (say) have the editor automatically scroll to the first error in a compilation buffer and/or to the corresponding point in the source code if known. It would be nice if we had similar functionality in helix.

**Comment by adrian-budau** [comment]
Are these not what you are looking for? https://docs.helix-editor.com/keymap.html#unimpaired (specifically the first 4 from the list).

**Reply by leungbk** [reply]
> Are these not what you are looking for? https://docs.helix-editor.com/keymap.html#unimpaired (specifically the first 4 from the list).

No, those are different.

**Comment by sudormrfbin** [comment]
If you have an LSP running you can use `]d` and `[d` to navigate between errors in the current file.

**Reply by leungbk** [reply]
> If you have an LSP running you can use `]d` and `[d` to navigate between errors in the current file.

As I mentioned in my other response, the functionality I brought up is distinct from moving between language-server diagnostics.

In Emacs, [`next-error` specifically uses compilation, grep, or occur (occur is similar to grep) buffers](https://www.gnu.org/software/emacs/manual/html_node/emacs/Compilation-Mode.html). You can move between compilation errors or grep matches.

So to put it another way, I am suggesting that Helix support a compilation feature similar to `M-x compile` in Emacs or `:make` in Vim, and accompanying compile-error navigation commands.

**Comment by kirawi** [comment, marked as answer]
https://github.com/helix-editor/helix/issues/2934
